lms/djangoapps/courseware/views/views.py

```diff
@@ -954,6 +949,13 @@ def _progress(request, course_key, student_id):
     if student_id is None or student_id == request.user.id:
         # This will be a no-op for non-staff users, returning request.user
         masquerade, student = setup_masquerade(request, course_key, staff_access, reset_masquerade_data=True)
+
+        # The pre-fetching of groups is done to make auth checks not require an
+        # additional DB lookup (this kills the Progress page in particular).
+        student = User.objects.prefetch_related("groups").get(id=student.id)
```

Comment:
Maybe we could use `prefetch_related_objects()`?

```python
masquerade, student = setup_masquerade(request, course_key, staff_access, reset_masquerade_data=True)
prefetch_related_objects(student, 'groups')
```

Reply:
this doesn't erase masquerade, but i'm not sure what the differece is between this prefetch line and the existing one.  
are we confident this prefetch line will still address the issue?

Reply:
The existing one will also re-fetch the user data. The version I wrote will only fetch the groups. I think you could replace both calls to `User.objects.` with this and move it outside of the if/else block.

Reply:
right, so I'm asking the question because I can see the queries are a bit different. so the comment for that line said it was necessary for some performance issue. are we confident that only fetching the groups and not refetching the user data will still fix the performance issue? if we need to that's something I can dig into, but since you wrote the line I was hoping you would have a sense of that

Reply:
If you look [at the masquerade code path](https://github.com/edx/edx-platform/blob/1a8df4517943a15b2d1c71be8ea44c4f6382d00c/common/djangoapps/student/models.py#L1157), it is basically doing the same thing you are doing here, so I think the resulting `User` object is the same regardless of the code path. In one case we just fetch the user data twice.

I would be surprised if we needed to re-fetch the user here, but I suppose that's theoretically possible?